### PR TITLE
add cluster quota APIs

### PR DESCRIPTION
**Pull request by deads2k**
The legacy openshift API alternative to https://github.com/openshift/origin/pull/9472

@openshift/api-review for comparison.  And the client "just works".

### CODE REVIEW

**Comment by deads2k** [comment]
[test]

File: pkg/quota/api/register.go [hunk 1]
```diff
@@ -30,6 +30,9 @@ func addKnownTypes(scheme *runtime.Scheme) {
 		&kapi.ListOptions{},
 		&ClusterResourceQuota{},
 		&ClusterResourceQuotaList{},
+
+		&kapi.DeleteOptions{},
+		&kapi.ListOptions{},
```
Comment (smarterclayton) [inline pkg/quota/api/register.go]: Add versioned watch if you haven't.

**Comment by smarterclayton** [comment]
Looks fine other than that.

File: pkg/quota/api/register.go (hunk 1 above)
Comment (deads2k) [inline pkg/quota/api/register.go]: > Add versioned watch if you haven't.

This pull puts clusterresourcequota in the legacy API group, which should already have that registered via kube.

File: pkg/cmd/server/origin/reststorage_validation_test.go [hunk 2]
```diff
@@ -66,22 +63,6 @@ func TestValidationRegistration(t *testing.T) {
 	}
 }
 
-// TestAllOpenShiftResourceCoverage checks to make sure that the openshift all group actually contains all openshift resources
-func TestAllOpenShiftResourceCoverage(t *testing.T) {
```
Comment (deads2k) [inline pkg/cmd/server/origin/reststorage_validation_test.go]: getting removed in https://github.com/openshift/origin/pull/9486

**Comment by smarterclayton** [comment]
LGTM once your tests pass.  Merge at will.

**Comment by deads2k** [comment]
[merge]

**Pull request merged**